.changeset/little-bulldogs-guess.md

```diff
@@ -0,0 +1,17 @@
+---
+'@backstage/cli': patch
+---
+
+Add ability to force use of the development versions of `react` and `react-dom`
+to allow for detailed error messages and use of fast-refresh. Note that builds
+with the development versions of `react` and `react-dom` should not be
+deployed in production.
+
+This feature can be used by setting the `FORCE_REACT_DEVELOPMENT` environment
+variable to `true`:
+
+```bash
+EXPERIMENTAL_MODULE_FEDERATION=true \
+FORCE_REACT_DEVELOPMENT=true \
+yarn build:all
+```
```

Comment:
```suggestion
The experimental module federation build now has the ability to force the use of development versions of `react` and `react-dom` by setting the `FORCE_REACT_DEVELOPMENT` flag.
```

Looking to keep this a little bit shorter to take up less space in the changelogs. This is still very experimental so I think it's best to not draw too much attention to it to minimize the risk of confusion.